### ISSUE TITLE
`reportMissingSuperCall` false positive when extending protocols

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#119]
```py
from typing import Protocol


class Foo(Protocol):
    pass

class Bar(Foo):
    def __init__(self) -> None:
        ...
```

**Comment by DetachHead** [comment]
actually this is an "as designed" moment, since protocols can be explicitly extended for some reason, so they have the same issue normal classes do with multiple inheritance
